Automattic/pocket-casts-ios
# Referrals: send pass share

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR adds the share capability for the Referrals send pass screen.

| 1 | 2 | 3 | 4 | 5 |
| - | - | - | - | - |
| ![IMG_0145](https://github.com/user-attachments/assets/61a59820-6596-43ba-bc4a-47f528ab06cb) |![IMG_0144](https://github.com/user-attachments/assets/ea559693-83c5-4584-afa3-e971471742e6) | ![IMG_0143](https://github.com/user-attachments/assets/8c3a35f8-5fd6-4162-bfe6-db4d8b8540da) | ![IMG_0141](https://github.com/user-attachments/assets/fc450bae-9c1f-47fb-8017-d467ecad48a5) | ![IMG_0142](https://github.com/user-attachments/assets/78e8ae26-73e5-412f-8ab3-dc9b331d08e7) |


I initially tried to wrap the UIActivityViewController on an SwiftUI View but was having issues with the presentation using a bottom sheet ( there was always a part of the bottom sheet that didn't get the right color)

I tried to user ShareLink but the functionality is iOS 16 and above only so I needed to replicate it using UIActivityViewController any way.

In the end I'm using the UIActivityViewController in the host VC for the SendPass View. This allow me to proper display it and keep it contained inside the SendPass flow.



## To test

1. Start the app
2. Ensure you the Referrals FF enabled and you are using a Plus or Patron account
3. Go to Profile
4. Tap on the gift icon on the top left
5. Tap On Share Guest Pass
6. See the Share system view
7. Choose different types of Share and check if they work:
8. Share and see if you go back to the profile screen.

## Checklist
